### PR TITLE
Add dependabot config to limited-partnerships-api

### PR DESCRIPTION
**Pull request by HatiDev**
### JIRA link

https://companieshouse.atlassian.net/browse/LP-168

### Change description

Add dependabot config to limited-partnerships-api

### Work checklist

* [ ] Tests added where applicable
* [ ] Bug fix
* [x] New feature
* [ ] Breaking change
* [ ] Infrastructure change

### Pull request checklist

* [ ] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__
### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# To get started with Dependabot version updates, you'll need to specify which
+# package ecosystems to update and where the package manifests are located.
+# Please see the documentation for all configuration options:
+# https://help.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  - package-ecosystem: "npm" # See documentation for possible values
```
Comment (markpit) [inline .github/dependabot.yml:8]: looks like you've copied the file from a node project instead of a java one, "npm" is for Nodejs

**Review by mattch1 — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (mwestacott) [reply .github/dependabot.yml:8]: Yep. Looks like this value should be "maven".

See https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem and https://github.com/companieshouse/overseas-entities-api/blob/main/.github/dependabot.yml for an example.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by markpit — APPROVED** [review]

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# To get started with Dependabot version updates, you'll need to specify which
+# package ecosystems to update and where the package manifests are located.
+# Please see the documentation for all configuration options:
+# https://help.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
```
[the remark this answers is not in the document]
Comment (HatiDev) [reply .github/dependabot.yml]: link updated

File: .github/dependabot.yml (hunk 1 above)
Comment (HatiDev) [reply .github/dependabot.yml]: package-ecosystem changed to maven for java dependencies

**Pull request merged**